Update storage format.

ilyalesokhin-starkware opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/starkware-libs/cairo/1594)
<!-- Reviewable:end -->

yuvalsw reviewed (changes requested):
> Reviewed 19 of 19 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 8 unresolved discussions (waiting on @ilyalesokhin-starkware, @liorgold2, and @orizi)

___
*[`Cargo.lock` line 231 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKiedCoERm92AsVpNwy:-NKiedCoERm92AsVpNwz:b-p16tu7) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/Cargo.lock#L231)):*
> ```TOML
>  "itertools",
>  "num-bigint",
>  "num-traits 0.2.15",
> ```

should the version be here?
___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKieim-4J5XxZMkih1q:-NKieim-4J5XxZMkih1r:b3tskr4) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
> ```Cairo
> extern fn storage_address_const<address>() -> StorageAddress nopanic;
> extern fn storage_read_syscall(
>     address_domain: felt, address: StorageAddress,
> ```

what is address domain?
___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKietfMCXUy9e6G4Cha:-NKietfMCXUy9e6G4Chb:buyd8lu) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
> ```Cairo
> extern fn storage_address_const<address>() -> StorageAddress nopanic;
> extern fn storage_read_syscall(
>     address_domain: felt, address: StorageAddress,
> ```

worth adding this change to the PR description.
___
*[`crates/cairo-casm/src/run.rs` line 193 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKifOez0b22G8PeBhE1:-NKifOez0b22G8PeBhE2:b-cn5j5c) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L193)):*
> ```XML
>                 };
>                 let (selector_sign, selector) =
>                     get_double_deref_val(cell, &base_offset)?.to_bytes_be();
> ```

why did you change to big endian?
___
*[`crates/cairo-casm/src/run.rs` line 234 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj02dQDojQtyAHQA6C:-NKj02dQDojQtyAHQA6D:bwwmzpt) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L234)):*
> ```XML
>                         vm.insert_value(&revert_reason_ptr, BigInt::from(0))?;
>                         vm.insert_value(&result_ptr, value)?;
>                     } else {
> ```

why is it a failure if addr_domain is 0? Can you please doc?
___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 92 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj7ooP6suyRERaNLhQ:-NKj7ooP6suyRERaNLhR:bm4skxf) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L92)):*
> ```XML
>                     ap_change: SierraApChange::Known { new_vars_only: false },
>                 },
>                 BranchSignature {
> ```

// Failure branch
___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 80 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj8sA05nE6dg9mjmYd:-NKj8sA05nE6dg9mjmYe:byiyy03) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L80)):*
> ```XML
>     let system = casm_builder.add_var(system);
>     let gas_builtin = casm_builder.add_var(ResOperand::Deref(gas_builtin));
>     let storage_address = casm_builder.add_var(ResOperand::Deref(storage_address));
> ```

can you please swap the order of the 2 to be consistent with their order in the params?



_Code quote:_
```XML
    let storage_address = casm_builder.add_var(ResOperand::Deref(storage_address));
    let address_domain = casm_builder.add_var(ResOperand::Deref(address_domain));
```
___
*[`crates/cairo-starknet/src/plugin.rs` line 313 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKjAEzB1kWH5eb0yDoD:-NKjAEzB1kWH5eb0yDoE:b1d4e9n) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-starknet/src/plugin.rs#L313)):*
> ```XML
>                     let address_domain = 0;
>                     match starknet::storage_read_syscall(
>                         address_domain,
> ```

can't you pass 0 directly here? Same below.
And why is this always 0?


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`Cargo.lock` line 231 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKiedCoERm92AsVpNwy:-NKkxiwa7-JTKJaia_-X:bpqy5o2) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/Cargo.lock#L231)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

should the version be here?
</blockquote></details>

I don't know this file is autogenerated by the toolchain.
___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKieim-4J5XxZMkih1q:-NKkxm9v6USnXGs_p6C2:b-lxvv2x) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

what is address domain?
</blockquote></details>

It is reserved for now.
I thought it would be better to add this way instead of writing 0 in the libfunc.

do you want me to rename it to reserved?
___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKietfMCXUy9e6G4Cha:-NKky4mdApDkGP8BTOre:bd6rhxl) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

worth adding this change to the PR description.
</blockquote></details>

this field is mostly reserved.
___
*[`crates/cairo-casm/src/run.rs` line 193 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKifOez0b22G8PeBhE1:-NKky7CZ6iVDWKVOKysW:bzdnh1y) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L193)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

why did you change to big endian?
</blockquote></details>

I was wrong about the endianness of short string.
___
*[`crates/cairo-casm/src/run.rs` line 234 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj02dQDojQtyAHQA6C:-NKkyD7y5p41auLJtQ2d:bwyagq5) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L234)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

why is it a failure if addr_domain is 0? Can you please doc?
</blockquote></details>

Because it is a reserved field.
___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 92 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj7ooP6suyRERaNLhQ:-NKkyYmn3b0wsROkYJzt:bz664ab) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L92)):*
> // Failure branch
done.
___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 80 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj8sA05nE6dg9mjmYd:-NKkyg0g-kcdany6y8dv:b-7rl0jd) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L80)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

can you please swap the order of the 2 to be consistent with their order in the params?
</blockquote></details>

done.



_Code quote:_
```XML
  let storage_address = casm_builder.add_var(ResOperand::Deref(storage_address));
```
___
*[`crates/cairo-starknet/src/plugin.rs` line 313 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKjAEzB1kWH5eb0yDoD:-NKkyimJ1REo4hISkO6G:b-59etfe) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-starknet/src/plugin.rs#L313)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

can't you pass 0 directly here? Same below.
And why is this always 0?
</blockquote></details>

I can; I thought it was clearer this way.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 8 of 19 files reviewed, 6 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 96 at r2](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKl70Yw3rPdYhVgLv5-:-NKl70Yw3rPdYhVgLv50:b-hhu266) ([raw file](https://github.com/starkware-libs/cairo/blob/14042d64aeec11f7ff6eba734b48a92b05572b2b/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L96)):*
> ```XML
>         // `revert_reason` is 0 on success, nonzero on failure/revert.
>         tempvar revert_reason = *(system++);
>         let _ignore = *(system++);
> ```

Does any one remember why we are advancing system here?



_Code quote:_
```XML
let _ignore = *(system++);
```


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 96 at r2](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKl70Yw3rPdYhVgLv5-:-NKl9lmf6nnr_i_PPMOG:b4m9zjd) ([raw file](https://github.com/starkware-libs/cairo/blob/14042d64aeec11f7ff6eba734b48a92b05572b2b/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L96)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

Does any one remember why we are advancing system here?
</blockquote></details>

Ok, looks like yoni's format is:

(storage_write)
REQUEST:
selector
gas
reserved
key
value
RESPONSE:
gas
result_selector
(if err) revert_reason



I wanted if we want to merge result_selector and revert_reason.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 8 of 20 files reviewed, 7 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 33 at r2](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlA3jTDGWhy5B65joN:-NKlA3jTDGWhy5B65joO:bhfq4bw) ([raw file](https://github.com/starkware-libs/cairo/blob/14042d64aeec11f7ff6eba734b48a92b05572b2b/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L33)):*
> ```XML
>     let storage_address = casm_builder.add_var(ResOperand::Deref(storage_address));
> 
>     casm_build_extend! {casm_builder,
> ```

(storage_read)
REQUEST:
selector
gas
key
reserved
RESPONSE:
gas
result_selector
(if ok) value
(if err) revert_reason



_Code quote:_
```XML
 casm_build_extend! {casm_builder,
```


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 33 at r2](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlA3jTDGWhy5B65joN:-NKlA9UKB5PRZSdewW6s:b1kjv39) ([raw file](https://github.com/starkware-libs/cairo/blob/14042d64aeec11f7ff6eba734b48a92b05572b2b/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L33)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

(storage_read)
REQUEST:
selector
gas
key
reserved
RESPONSE:
gas
result_selector
(if ok) value
(if err) revert_reason
</blockquote></details>

I moved the reserved field and changed key -> address.


REQUEST:
selector
gas
address_domain
address

RESPONSE:
gas
result_selector
(if ok) value


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 8 of 20 files reviewed, 7 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 96 at r2](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKl70Yw3rPdYhVgLv5-:-NKlAPQ96tz04Vr7C4cs:b7hg43l) ([raw file](https://github.com/starkware-libs/cairo/blob/14042d64aeec11f7ff6eba734b48a92b05572b2b/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L96)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

Ok, looks like yoni's format is:

(storage_write)
REQUEST:
selector
gas
reserved
key
value
RESPONSE:
gas
result_selector
(if err) revert_reason



I wanted if we want to merge result_selector and revert_reason.

</blockquote></details>

I moved the reserved field and changed key -> address.

storage_write)
REQUEST:
selector
gas
address_domain
address
value
RESPONSE:
gas
result_selector


<!-- Sent from Reviewable.io -->

orizi reviewed (changes requested):
> Reviewed 3 of 19 files at r1, 5 of 11 files at r2, 3 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 16 of 20 files reviewed, 8 unresolved discussions (waiting on @ilyalesokhin-starkware, @liorgold2, and @yuvalsw)

___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 147 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWOeG99wI-bIKL43r:-NKlWOeG99wI-bIKL43s:b-ykrcwu) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L147)):*
> ```XML
>                 },
>                 // Address domain
>                 ParamSignature::new(felt_ty.clone()),
> ```

which sort of values are these gonna be?
i understand 0 is for regular addresses - what are other options? only 1 for forced transactions?



_Code quote:_
```XML
                // Address domain
                ParamSignature::new(felt_ty.clone()),
```
___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs` line 15 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWmVgBWI9S_Ck560c:-NKlWmVgBWI9S_Ck560d:b58p51m) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs#L15)):*
> ```XML
> 
> #[test]
> fn test_storage_read() {
> ```

consider moving these to e2e libfunc tests



_Code quote:_
```XML
test_storage_read() {
```


<!-- Sent from Reviewable.io -->

orizi reviewed (commented):
> Reviewed 3 of 11 files at r2, 1 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 8 unresolved discussions (waiting on @ilyalesokhin-starkware, @liorgold2, and @yuvalsw)



<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 147 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWOeG99wI-bIKL43r:-NKlZPH130cPK8C5gSdU:b-sw586x) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L147)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

which sort of values are these gonna be?
i understand 0 is for regular addresses - what are other options? only 1 for forced transactions?
</blockquote></details>

The address domains are for different data availability guarantees.
Today we only support onchain data ability.
We want to add cheaper options in the future.
Lior wanted it to be a parameter rather than a separate syscall.



Do you want me to rename this field reserved?
I prefer to to set it to 0 in the libfunc as it requires another temp var.
This way the 0 tempvar can be shared between calls.
___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs` line 15 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWmVgBWI9S_Ck560c:-NKlZIFC0KhLbW_Utxt4:blegixd) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs#L15)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

consider moving these to e2e libfunc tests
</blockquote></details>

future PR, this one is large enough.


<!-- Sent from Reviewable.io -->

orizi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 19 files at r1.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594#-:-NKlbilr4Ft4-Gpvwe0s:bnfp4nl)* status: all files reviewed, 6 unresolved discussions (waiting on @ilyalesokhin-starkware, @liorgold2, and @yuvalsw)

___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 147 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWOeG99wI-bIKL43r:-NKlbTsJ1JosDZ6jCgGu:b3zmpih) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L147)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

The address domains are for different data availability guarantees.
Today we only support onchain data ability.
We want to add cheaper options in the future.
Lior wanted it to be a parameter rather than a separate syscall.



Do you want me to rename this field reserved?
I prefer to to set it to 0 in the libfunc as it requires another temp var.
This way the 0 tempvar can be shared between calls.
</blockquote></details>

was mostly wondering if this should be some other datatype - but sounds good enough.


<!-- Sent from Reviewable.io -->

liorgold2 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @ilyalesokhin-starkware and @yuvalsw)

___
*[`crates/cairo-starknet/test_data/hello_starknet.casm` line 1 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlfu6R8b3DzXxs1KfR:-NKlfu6R8b3DzXxs1KfS:b-xifkwm) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-starknet/test_data/hello_starknet.casm#L1)):*
> ```
> {
> ```

This is not a `casm` file, it's a `json` file, please rename in a different PR.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @yuvalsw)

___
*[`crates/cairo-starknet/test_data/hello_starknet.casm` line 1 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlfu6R8b3DzXxs1KfR:-NKlgdoX8g_hoVkp78Z0:b-d6pj3b) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-starknet/test_data/hello_starknet.casm#L1)):*
<details><summary><i>Previously, liorgold2 wrote…</i></summary><blockquote>

This is not a `casm` file, it's a `json` file, please rename in a different PR.
</blockquote></details>

it is a casm contract.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @orizi and @yuvalsw)

___
*[`crates/cairo-sierra/src/extensions/modules/starknet/storage.rs` line 147 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlWOeG99wI-bIKL43r:-NKlgpHjAMxrs61pD2pv:b6xcm6s) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra/src/extensions/modules/starknet/storage.rs#L147)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

was mostly wondering if this should be some other datatype - but sounds good enough.
</blockquote></details>

I've considered it.

But It is more work, and I think the OS is going to branch based on this value, so it might as well return an error when it is out of range.


<!-- Sent from Reviewable.io -->

liorgold2 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 7 unresolved discussions (waiting on @ilyalesokhin-starkware, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs` line 32 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlhYF83Qaxgd0y8D8v:-NKlhYF83Qaxgd0y8D8w:b-6b8ug4) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs#L32)):*
> ```XML
>         ReducedCompiledInvocation {
>             instructions: casm! {
>                 [ap + 0] = 35731537817157092826019094884u128, ap++;
> ```

In the past we use CamelCase. Not that it's really matter, just making sure we're aware of this change.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 7 unresolved discussions (waiting on @orizi and @yuvalsw)

___
*[`crates/cairo-starknet/test_data/hello_starknet.casm` line 1 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlfu6R8b3DzXxs1KfR:-NKljQG4AoHBb3VrMfxV:brw7a9i) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-starknet/test_data/hello_starknet.casm#L1)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

it is a casm contract.
</blockquote></details>

what name are you suggesting?


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 7 unresolved discussions (waiting on @liorgold2, @orizi, and @yuvalsw)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs` line 32 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlhYF83Qaxgd0y8D8v:-NKlkb3sCKXPep9ECMfE:b-lxd5rz) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs#L32)):*
<details><summary><i>Previously, liorgold2 wrote…</i></summary><blockquote>

In the past we use CamelCase. Not that it's really matter, just making sure we're aware of this change.
</blockquote></details>

Do you want me to change it to camel case?


<!-- Sent from Reviewable.io -->

orizi reviewed (approved):
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 6 unresolved discussions (waiting on @liorgold2 and @yuvalsw)



<!-- Sent from Reviewable.io -->

yuvalsw reviewed (approved):
> Reviewed 7 of 11 files at r2, 1 of 4 files at r3, 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 5 unresolved discussions (waiting on @ilyalesokhin-starkware and @liorgold2)

___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKieim-4J5XxZMkih1q:-NKlsvHsAX1W1DdH7E_O:bg4de57) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

It is reserved for now.
I thought it would be better to add this way instead of writing 0 in the libfunc.

do you want me to rename it to reserved?
</blockquote></details>

It would be best if there would be some documentation of what it is expected to be. I don't know what an address domain is. If it is reserved specifically for it, only add doc about what it is and that it's reserved. If we don't know how we are going to use it, then reserved is ok (although I don't think we should add a reserved param to each syscall...)
___
*[`crates/cairo-casm/src/run.rs` line 234 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj02dQDojQtyAHQA6C:-NKlwPnx9sCHrRiZtJh1:bfuhg7b) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L234)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

Because it is a reserved field.
</blockquote></details>

please doc.
___
*[`crates/cairo-casm/src/run.rs` line 195 at r4](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKltR5GDJzSzRPgmBjZ:-NKltR5GDJzSzRPgmBj_:byglrpl) ([raw file](https://github.com/starkware-libs/cairo/blob/ab5f70a20a92b2ddcc3b0b6afcf4578a14ba69b8/crates/cairo-casm/src/run.rs#L195)):*
> ```XML
>                     get_double_deref_val(cell, &base_offset)?.to_bytes_be();
>                 assert_eq!(selector_sign, num_bigint::Sign::Plus, "Illegal selector.");
>                 if selector == "StorageWrite".as_bytes() {
> ```

is this the way it's in the os?



_Code quote:_
```XML
StorageWrite
```
___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 71 at r4](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlzl7d6InFYKaZWW51:-NKlzl7eDaBgFKui4W2s:bmuuh2l) ([raw file](https://github.com/starkware-libs/cairo/blob/84f4781e92278b9479062f3ef4e5d7b3c9796e7d/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L71)):*
> ```XML
>         // `revert_reason` is 0 on success, nonzero on failure/revert.
>         tempvar revert_reason = *(system++);
>         let _ignore = *(system++);
> ```

why was this removed?
___
*[`crates/cairo-starknet/src/plugin.rs` line 313 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKjAEzB1kWH5eb0yDoD:-NKlzzzrATOnV6j2iKBy:bebkuhj) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-starknet/src/plugin.rs#L313)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I can; I thought it was clearer this way.
</blockquote></details>

Hmmm I am not sure. Unless you call it reserved_address_domain/reserved or doc it (in the generated code) saying it's reserved.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 71 at r4](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlzl7d6InFYKaZWW51:-NKm1l6OCFYkcQvItckS:b-9kf0ho) ([raw file](https://github.com/starkware-libs/cairo/blob/84f4781e92278b9479062f3ef4e5d7b3c9796e7d/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L71)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

why was this removed?
</blockquote></details>

we decided to merge

is_reverted and revert_reason.



_Code quote:_
```XML
revert_reason
```


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware commented:
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs` line 32 at r3](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlhYF83Qaxgd0y8D8v:-NKm1oaf6EEuHPfOJq87:b-7rl0jd) ([raw file](https://github.com/starkware-libs/cairo/blob/b8c6152b0ca2435373d021f2aa13cef31b7cec9a/crates/cairo-sierra-to-casm/src/invocations/starknet/storage_test.rs#L32)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

Do you want me to change it to camel case?
</blockquote></details>

done.


<!-- Sent from Reviewable.io -->

yuvalsw reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: all files reviewed, 4 unresolved discussions (waiting on @ilyalesokhin-starkware)

___
*[`crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs` line 71 at r4](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKlzl7d6InFYKaZWW51:-NKm2T2qDh2V-cSme2SM:buwv5ym) ([raw file](https://github.com/starkware-libs/cairo/blob/84f4781e92278b9479062f3ef4e5d7b3c9796e7d/crates/cairo-sierra-to-casm/src/invocations/starknet/storage.rs#L71)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

we decided to merge

is_reverted and revert_reason.
</blockquote></details>

this ignore wasn't for is_reverted. We merged them in the first place... (I don't remember what the ignore was for though)


<!-- Sent from Reviewable.io -->

yuvalsw reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594#-:-NKm2uaJE3wQguzxovZj:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @ilyalesokhin-starkware)



<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 14 of 20 files reviewed, 2 unresolved discussions (waiting on @orizi and @yuvalsw)

___
*[`crates/cairo-casm/src/run.rs` line 234 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKj02dQDojQtyAHQA6C:-NKm2n42D8FyUeFu2_4A:b-7rl0jd) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/crates/cairo-casm/src/run.rs#L234)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

please doc.
</blockquote></details>

done.
___
*[`crates/cairo-casm/src/run.rs` line 195 at r4](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKltR5GDJzSzRPgmBjZ:-NKm1dzi5UDYh7f7aPc1:b-ggl1hy) ([raw file](https://github.com/starkware-libs/cairo/blob/ab5f70a20a92b2ddcc3b0b6afcf4578a14ba69b8/crates/cairo-casm/src/run.rs#L195)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

is this the way it's in the os?
</blockquote></details>

I hope so.


<!-- Sent from Reviewable.io -->

ilyalesokhin-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: 14 of 20 files reviewed, 2 unresolved discussions (waiting on @orizi and @yuvalsw)

___
*[`corelib/starknet.cairo` line 12 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1594#-NKieim-4J5XxZMkih1q:-NKm3XFv4JxJOrpVJ65v:bt4zlpe) ([raw file](https://github.com/starkware-libs/cairo/blob/580ca416225b8ca80e96b1971523f2b5b1d3a637/corelib/starknet.cairo#L12)):*
<details><summary><i>Previously, yuvalsw wrote…</i></summary><blockquote>

It would be best if there would be some documentation of what it is expected to be. I don't know what an address domain is. If it is reserved specifically for it, only add doc about what it is and that it's reserved. If we don't know how we are going to use it, then reserved is ok (although I don't think we should add a reserved param to each syscall...)
</blockquote></details>

I'm not sure where to put the documentation. 


<!-- Sent from Reviewable.io -->

orizi reviewed (approved):
> Reviewed 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ilyalesokhin-starkware)



<!-- Sent from Reviewable.io -->

orizi reviewed (approved):
> Reviewed 5 of 6 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1594)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ilyalesokhin-starkware)



<!-- Sent from Reviewable.io -->
